subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/classpath/ClasspathEntrySnapshotCache.java

```diff
@@ -25,7 +25,7 @@
 public interface ClasspathEntrySnapshotCache extends Cache<File, ClasspathEntrySnapshot> {
     /**
      * Returns the classpath entry snapshots for the given files. The resulting map has the same order as the input.
-     * Fails if no cached data could be found for any of the given files.
+     * Some files may not have cached data and they will be missing from the returned result.
```

Comment:
There was actually no reason to fail here. If the previous data is missing, then all classes in that jar will be considered "added" and thus out-of-date.